### ISSUE TITLE
Latest vassal dev builds giving Invalid Proxy exception

### ISSUE THREAD

**Issue by riverwanderer** [vassalengine/vassal#12935]
To repro take a recent build. I was using VASSAL-3.7.6-SNAPSHOT-e2c7e59-master-macos-universal.
Start any module - doesn't matter what.
Connect to the Vassal server via Server Controls **connect** button.
An exception is generated immediately.

Error log: 
```
2023-11-27 21:12:57,152 [3314-main] INFO  VASSAL.launch.StartUp - Starting
2023-11-27 21:12:57,161 [3314-main] INFO  VASSAL.launch.StartUp - OS Mac OS X 14.1.1 x86_64
2023-11-27 21:12:57,162 [3314-main] INFO  VASSAL.launch.StartUp - Java version 21
2023-11-27 21:12:57,162 [3314-main] INFO  VASSAL.launch.StartUp - Java home /Applications/VASSAL 2.app/Contents/MacOS/jre
2023-11-27 21:12:57,162 [3314-main] INFO  VASSAL.launch.StartUp - VASSAL version 3.7.6-SNAPSHOT-e2c7e59-master
2023-11-27 21:12:57,162 [3314-main] INFO  VASSAL.launch.Launcher - Player
2023-11-27 21:12:58,808 [3314-AWT-EventQueue-0] INFO  VASSAL.build.GameModule - Cat & MouseOver Not OK version 0.0
2023-11-27 21:13:09,677 [3314-AWT-EventQueue-0] ERROR VASSAL.chat.node.NodeClientFactory - Unable to establish connection to server: Network is unreachable
2023-11-27 21:13:33,071 [3314-AWT-EventQueue-0] ERROR VASSAL.chat.node.NodeClientFactory - Unable to establish connection to server: Network is unreachable
2023-11-27 21:13:36,866 [3314-AWT-EventQueue-0] ERROR VASSAL.chat.node.NodeClientFactory - Unable to establish connection to server: game.vassalengine.org
2023-11-27 21:13:38,283 [3314-AWT-EventQueue-0] ERROR VASSAL.chat.node.NodeClientFactory - Unable to establish connection to server: game.vassalengine.org
2023-11-27 21:13:39,282 [3314-AWT-EventQueue-0] ERROR VASSAL.chat.node.NodeClientFactory - Unable to establish connection to server: game.vassalengine.org
2023-11-27 21:13:55,772 [3314-AWT-EventQueue-0] ERROR VASSAL.chat.node.NodeClientFactory - Unable to establish connection to server: game.vassalengine.org
2023-11-27 21:16:53,386 [3314-AWT-EventQueue-0] ERROR VASSAL.tools.ErrorDialog - 
java.lang.IllegalArgumentException: Invalid Proxy
	at java.base/java.net.Socket.<init>(Socket.java:216)
	at org.apache.hc.client5.http.ssl.SSLConnectionSocketFactory.createSocket(SSLConnectionSocketFactory.java:208)
	at org.apache.hc.client5.http.impl.io.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:158)
	at org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:447)
	at org.apache.hc.client5.http.impl.classic.InternalExecRuntime.connectEndpoint(InternalExecRuntime.java:162)
	at org.apache.hc.client5.http.impl.classic.InternalExecRuntime.connectEndpoint(InternalExecRuntime.java:172)
	at org.apache.hc.client5.http.impl.classic.ConnectExec.execute(ConnectExec.java:142)
	at org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
	at org.apache.hc.client5.http.impl.classic.ProtocolExec.execute(ProtocolExec.java:192)
	at org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
	at org.apache.hc.client5.http.impl.classic.HttpRequestRetryExec.execute(HttpRequestRetryExec.java:96)
	at org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
	at org.apache.hc.client5.http.impl.classic.ContentCompressionExec.execute(ContentCompressionExec.java:152)
	at org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
	at org.apache.hc.client5.http.impl.classic.RedirectExec.execute(RedirectExec.java:115)
	at org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
	at org.apache.hc.client5.http.impl.classic.InternalHttpClient.doExecute(InternalHttpClient.java:170)
	at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:123)
	at VASSAL.chat.HttpRequestWrapper.doGet(HttpRequestWrapper.java:101)
	at VASSAL.chat.HttpRequestWrapper.doGet(HttpRequestWrapper.java:54)
	at VASSAL.chat.HttpMessageServer.getWelcomeMessage(HttpMessageServer.java:72)
	at VASSAL.chat.node.NodeClient.setConnected(NodeClient.java:163)
	at VASSAL.chat.HybridClient.setConnected(HybridClient.java:90)
	at VASSAL.chat.DynamicClient.access$001(DynamicClient.java:39)
	at VASSAL.chat.DynamicClient$1.done(DynamicClient.java:64)
	at java.desktop/javax.swing.SwingWorker$4.run(SwingWorker.java:749)
	at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(SwingWorker.java:847)
	at java.desktop/sun.swing.AccumulativeRunnable.run(AccumulativeRunnable.java:112)
	at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(SwingWorker.java:857)
	at java.desktop/javax.swing.Timer.fireActionPerformed(Timer.java:311)
	at java.desktop/javax.swing.Timer$DoPostEvent.run(Timer.java:243)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

```

**Comment by riverwanderer** [comment]
VASSAL-3.7.6-SNAPSHOT-0a681bb-deck_sortListener_reset-macos-universal from Nov 20th is OK.

**Comment by riverwanderer** [comment]
@uckelman diagnosed the problem in this [discord thread](https://discord.com/channels/740127632586178572/740127633215586337/1178826197099085906).

External (Apache) -to be fixed in H[TTPCLIENT-2292 5.2.3.](https://issues.apache.org/jira/browse/HTTPCLIENT-2292?focusedCommentId=17789846&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17789846)
